bitcoin/src/bip32.rs

```diff
@@ -718,19 +768,21 @@ impl Xpub {
     /// the internal public key representation.
     pub fn to_x_only_pub(self) -> XOnlyPublicKey { XOnlyPublicKey::from(self.public_key) }
 
-    /// Attempts to derive an extended public key from a path.
+    /// Derives an extended public key from a path.
     ///
     /// The `path` argument can be any type implementing `AsRef<ChildNumber>`, such as `DerivationPath`, for instance.
     pub fn derive_pub<C: secp256k1::Verification, P: AsRef<[ChildNumber]>>(
```

Comment:
Maybe this should take a:

```suggestion
    pub fn derive_pub<C: secp256k1::Verification, P: AsRef<[NormalChildNumber]>>(
```
?